File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/scala/redocpot/spark/StrCount.scala
package scala.redocpot.spark

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object StrCount {
  def hasStr(s : String)(line : String) : Boolean = {
    line.contains(s)
  }

  def main(args : Array[String]) : Unit = {
    val conf = new SparkConf().setAppName("StrCount").setMaster("local[*]")
    val sc = new SparkContext(conf)
    val textFile = sc.textFile("/Users/me/Code/spark-learning/learning-spark-lightning-fast-big-data-analysis/pom.xml")
    //val xmlLines = textFile.filter(line => line.contains("xml"))
    val xmlLines = textFile.filter(hasStr("xml"))
    println("Input had " + xmlLines.count() + " concerning lines")
    println("Here are some examples:")
    xmlLines.take(10).foreach(println)
    sc.stop
  }
}

File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/java/redocpot/spark/StrCount.java
package redocpot.spark;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class StrCount {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
            .setMaster("local")
            .setAppName("StrCount");
        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> textFile = sc.textFile(
                "/Users/me/Code/spark-learning/learning-spark-lightning-fast-big-data-analysis/pom.xml");
        JavaRDD<String> xmlLines = textFile.filter(line -> line.contains("xml"));
        System.out.println("Input had " + xmlLines.count() + " concerning lines");
        System.out.println("Here are some examples:");
        for (String line: xmlLines.take(10)) {
            System.out.println(line);
        }

        sc.stop();
    }
}
